Tools/survey_agent.py

```diff
@@ -0,0 +1,113 @@
+from typing import Dict, Optional
+from Constants.prompts import SURVEY_DONT_UNDERSTAND_PROMPT
+from Tools.survey_helper import ask_question, create_user_details_from_answers
+
+
+class SurveyState:
+    """Class to manage survey state for a user."""
+
+    def __init__(self):
+        self.current_question: int = 0
+        self.answers: Dict[str, str] = {}
+        self.awaiting_confirmation: bool = False
+
+
+# Survey questions
+QUESTIONS = [
+    "Jaka jest Twoja waga?",
+    "Jaki jest Twój rok urodzenia?",
+    "Jaka jest Twoja płeć?",
+    "Czy masz alergie? Jeśli tak, proszę wymienić je."
+]
+
+# Store survey state for each user
+survey_states: Dict[str, SurveyState] = {}
+
+
+def get_survey_state(user_id: str) -> Optional[SurveyState]:
+    """Get survey state for a user if it exists."""
+    return survey_states.get(user_id)
+
+
+def create_survey_state(user_id: str) -> SurveyState:
+    """Create new survey state for a user."""
+    survey_states[user_id] = SurveyState()
+    return survey_states[user_id]
+
+
+def clear_survey_state(user_id: str) -> None:
+    """Clear survey state for a user."""
+    survey_states.pop(user_id, None)
+
+
+def get_or_create_survey_state(user_id: str) -> SurveyState:
```

Comment:
Method name should be get_survey_state. The create logic is mentioned in the comment. That's enough.